### PR TITLE
Add: webdav sync

### LINKED ISSUE #1001: [Feature Request] Sync to cloud: Sync by WebDAV
* **Browser**: Cent Browser 4.2.10.171 (Chromium 80.0.3987.163) (64-bit), Chrome 84.0.4147.89 (Official Build) (64-bit)
* **Operating System**: Windows 10  1909
* **Stylus Version**: 1.5.11

Users can use WebDAV on any WebDAV server such as Box / Nutstore (jianguoyun) / build by themselves.

### PR DESCRIPTION
**Pull request by eight04**
Fixes #1001

### CODE REVIEW

**Comment by eight04** [comment]
image

File: background/sync-manager.js [hunk 1]
```diff
@@ -85,19 +86,32 @@ const syncMan = (() => {
       return ctrl.put(...args);
     },
 
+    async setDriveOptions(driveName, options) {
+      const key = `secure/sync/driveOptions/${driveName}`;
+      await browser.storage.sync.set({
```
Comment (tophf) [inline background/sync-manager.js:91]: chromeSync.setValue(key, options)

and add chromeSync to `global` comment after chromeLocal.

File: background/sync-manager.js [hunk 2]
```diff
@@ -85,19 +86,32 @@ const syncMan = (() => {
...
+        [key]: options,
+      });
+    },
+
+    async getDriveOptions(driveName) {
+      const key = `secure/sync/driveOptions/${driveName}`;
+      const r = await browser.storage.sync.get(key);
+      return r[key] || {};
```
Comment (tophf) [inline background/sync-manager.js:99]: return chromeSync.getValue(key)

File: options/options.css [hunk 3]
```diff
@@ -437,6 +437,11 @@ input[type="radio"].radio:checked::after {
 .sync-status::first-letter {
   text-transform: uppercase;
 }
+.sync-options .drive-options {
+  margin: 0;
+  padding: 0;
+  border: 0;
+}
```
Comment (tophf) [inline options/options.css:445]:
```css
.webdav-options {
  display: table;
  width: 100%;
}
.webdav-options > label {
  display: table-row;
}
.webdav-options > label > * {
  display: table-cell;
}
.webdav-options input {
  width: 100%;
  box-sizing: border-box;
}
```

Comment (tophf) [reply options/options.css:445]: Also line 166 should be:
```css
label:not([disabled]) :not([type="number"]):not([type="text"]) {
```

**Comment by eight04** [comment]
Done

File: background/sync-manager.js [hunk 4]
```diff
@@ -88,15 +88,12 @@ const syncMan = (() => {
...
-        [key]: options,
-      });
+      await chromeSync.setValue(key, options);
     },
 
     async getDriveOptions(driveName) {
       const key = `secure/sync/driveOptions/${driveName}`;
-      const r = await browser.storage.sync.get(key);
-      return r[key] || {};
+      return chromeSync.getValue(key) || {};
```
Comment (tophf) [inline background/sync-manager.js:96]: This should have `await` in order for `||` to work:
```js
return await chromeSync.getValue(key) || {};
```

Comment (eight04) [reply background/sync-manager.js]: Good catch. BTW I didn't know `await` has the same precedence as `!` until now.

**Pull request merged**